### PR TITLE
completions: [git] Don't try to complete files when not in a repo

### PR DESCRIPTION
**Pull request by gechr**
Quick fix for: 

image

### CODE REVIEW

**Comment by gechr** [comment]
Oops, fixed.

> It might actually be nice, if you're not in a git repo, to just offer git init, git -C and such.

I often use `git diff` outside of a Git repository as I find it superior to `diff`. I also sometimes just use tab-completion to quickly remind myself of the available options. We should be careful when presuming how different people might use these completions and not try to be too smart about it.

File: share/completions/git.fish [hunk 1]
```diff
@@ -69,6 +73,9 @@ function __fish_git_files
     # and as a convenience "all-staged"
     # to get _all_ kinds of staged files.
 
+    # Do not continue if not inside a Git repository
+    __fish_git_is_worktree; or return
```
Comment (faho) [inline share/completions/git.fish]: Why not just 

```fish
set -l root (command git rev-parse --show-toplevel 2>/dev/null)
or return
```

?

I've never really worked with multiple work trees and such, so I'm not quite sure if `--is-inside-work-tree` is actually useful.

**Comment by faho** [comment]
>I often use git diff outside of a Git repository as I find it superior to diff

Huh... I did not know that works!

>We should be careful when presuming how different people might use these completions

Sure. What I mean is if something is objectively useless - like `git log` outside of a git repo. That one will just error out.

Anyway, it's not all that important.

File: share/completions/git.fish (hunk 1 above)
Comment (gechr) [reply share/completions/git.fish]: Fair enough.

It also just occurred to me that `--is-inside-work-tree` will return non-zero for bare repositories, which is undesirable.

**Pull request merged**